core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -107,4 +112,47 @@ public static boolean ancestorOf(Table table, long snapshotId, long ancestorSnap
 
     return newFiles;
   }
+
+  public static List<Snapshot> snapshotsWithin(Table table, long fromSnapshotId, long toSnapshotId) {
```

Comment:
Might it be good to add some javadoc here? I could see someone calling this when they really want `snapshotIdsBetween`

Reply:
Ah yeah I extracted these methods from IncrementalTableScan where they were private, now that they have been more exposed they need docs

Reply:
Can we put a TODO for the doc or potentially mark this protected? I don't want to block your PR because you extracted it from elsewhere where it was already private, but it would be nice to have a way to mark this with `// Consider snapshotIdsBetween` in one way or another. But I don't want to block your PR so if you'll come back to it later then that's ok.